### PR TITLE
added archive/unarchive feature

### PR DESCRIPTION
**Pull request by akorneyc**
This PR include:
-added archive implementation
-added unarchive implementation
-added unit and postman tests
-exported model, sql and png from pgmodeller

### CODE REVIEW

File: doc/dokuti.sql [hunk 1]
```diff
@@ -28,13 +28,14 @@ SET search_path TO pg_catalog,public,_documents;
...
 -- DROP TABLE IF EXISTS _documents.document CASCADE;
-CREATE TABLE _documents.document(
+CREATE TABLE _documents.document (
 	id uuid NOT NULL,
 	name text NOT NULL,
 	description text,
 	content_type text,
 	__updated_on timestamp NOT NULL,
 	__updated_by uuid NOT NULL,
+	is_archive boolean NOT NULL DEFAULT false,
```
Comment (melissapalmer) [inline doc/dokuti.sql]: I may have named this `is_archieved` rather.

File: doc/dokuti_api.yml [hunk 2]
```diff
@@ -1793,6 +1793,9 @@ components:
         description:
           type: string
           example: A test description for the file.
+        isArchive:
```
Comment (melissapalmer) [inline doc/dokuti_api.yml]: Same with naming above I'd have used `isArchived`

File: doc/dokuti_api.yml [hunk 3]
```diff
@@ -1793,6 +1793,9 @@ components:
...
+          type: boolean
+          example: true
```
Comment (melissapalmer) [inline doc/dokuti_api.yml]: Default on DB is false, I would default value here to false as well.

File: src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java [hunk 4]
```diff
@@ -240,4 +240,24 @@ private DocumentAttribute createOrUpdateDocumentAttribute(UUID documentId, Integ
...
+        DocumentEntity documentEntity = documentService.findById(documentId);
+        documentEntity.setIsArchive(status);
+        documentEntity = documentService.save(documentEntity);
+        Document res = databaseEntityToApiDataTranfserObjectMapperService.mapDocumentEntityToDocument(documentEntity);
+        return new ResponseEntity<>(res, HttpStatus.OK);    
+
+    }
+	
+    @Override
+    public ResponseEntity<Document> aricheveDocument(UUID documentId) {
```
Comment (melissapalmer) [inline src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java]: Spelling error `archive` not  `aricheve`

File: src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java [hunk 5]
```diff
@@ -240,4 +240,24 @@ private DocumentAttribute createOrUpdateDocumentAttribute(UUID documentId, Integ
 		Document res = databaseEntityToApiDataTranfserObjectMapperService.mapDocumentEntityToDocument(documentEntity);
 		return new ResponseEntity<>(res, HttpStatus.OK);			
     }	
+	
+    private ResponseEntity<Document> changeAricheveStatus(UUID documentId, Boolean status) {
```
Comment (melissapalmer) [inline src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java]: Spelling error `archive` not `aricheve`

File: src/test/java/za/co/grindrodbank/dokuti/document/DocumentControllerUnitTests.java [hunk 6]
```diff
@@ -125,4 +143,66 @@ public void givenDocument_whenPutDocumentInfo_thenReturnUpdatedInfo() throws Exc
 				  .characterEncoding("utf-8"))
 		          .andExpect(status().is(200));	 
 	}
+	
+    @Test
+	public void givenDocument_whenDoDocumentArchive_thenReturnArchevedDocumnet() throws Exception {
+
```
Comment (melissapalmer) [inline src/test/java/za/co/grindrodbank/dokuti/document/DocumentControllerUnitTests.java]: Please format code correctly/nicely ...

File: doc/dokuti.sql (hunk 1 above)
Comment (akorneyc) [reply doc/dokuti.sql]: fixed

File: doc/dokuti_api.yml (hunk 2 above)
Comment (akorneyc) [reply doc/dokuti_api.yml]: fixed

File: doc/dokuti_api.yml (hunk 3 above)
Comment (akorneyc) [reply doc/dokuti_api.yml]: fixed

File: src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java (hunk 4 above)
Comment (akorneyc) [reply src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java]: fixed

File: src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java (hunk 5 above)
Comment (akorneyc) [reply src/main/java/za/co/grindrodbank/dokuti/document/DocumentControllerImpl.java]: fixed

File: src/test/java/za/co/grindrodbank/dokuti/document/DocumentControllerUnitTests.java (hunk 6 above)
Comment (akorneyc) [reply src/test/java/za/co/grindrodbank/dokuti/document/DocumentControllerUnitTests.java]: fixed

**Pull request merged**